### PR TITLE
#710 lesson controller to what panel

### LINKED ISSUE #710: Add LessonController to WHAT Panel

### LINKED ISSUE #740: Add opportunity to check presence at the lesson and put marks (WHAT panel)

### PR DESCRIPTION
**Pull request by maksymiu4e**

### CODE REVIEW

**Review by AccorDDD — APPROVED** [review]
All looks good, but on my mind too many changes for one issue

**Review by leshikevich — APPROVED** [review]

File: CharlieBackend.Panel/Services/StudentGroupService.cs [hunk 1]
```diff
@@ -43,8 +46,14 @@ public async Task<IList<StudentGroupViewModel>> GetAllStudentGroupsAsync()
             var studentGroupsResponse = await _apiUtil.GetAsync<IList<StudentGroupDto>>(getAllStudentGroupsEndpoint);
             var studentGroups = _mapper.Map<IList<StudentGroupViewModel>>(studentGroupsResponse);
             
-            var students = await _studentService.GetAllStudentsAsync(); ;
-            var mentors = await _mentorService.GetAllMentorsAsync();
+            var students = await _studentService.GetAllStudentsAsync();
+
+            var mentors = await _mentorService.GetAllActiveMentorsAsync();
+            if (!(_currentUserService.Role == Core.Entities.UserRole.Mentor))
```
Comment (BobTheSuperman) [inline CharlieBackend.Panel/Services/StudentGroupService.cs]: Not a very clear condition, in my opinion it's more understandable
if (_currentUserService.Role != Core.Entities.UserRole.Mentor)

File: CharlieBackend.Panel/Views/Lessons/Attendance.cshtml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                                        @if (@Model.Visit[i].Presence)
+                                        {
+                                            <td class="col-2 text-truncate">+</td>
+
+                                        }
+                                        @if (!@Model.Visit[i].Presence)
+                                        {
+                                            <td class="col-2 text-truncate">-</td>
+
+                                        }
```
Comment (BobTheSuperman) [inline CharlieBackend.Panel/Views/Lessons/Attendance.cshtml]: I think it's better to use an if else construction than two if

File: CharlieBackend.Panel/Views/Lessons/Events.cshtml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                                    <th class="col-2">Theme</th>
+                                    <th class="col-2">Mentor</th>
+                                    <th class="col-2">Start</th>
+                                    <th class="col-2">Finish</th>
+                                    <th class="col-1">Controll</th>
+                                </tr>
+                            </thead>
+
+                            <tbody>
+                                @foreach (var item in @Model)
```
Comment (BobTheSuperman) [inline CharlieBackend.Panel/Views/Lessons/Events.cshtml]: It is not clear from the name what item stores in itself.

File: CharlieBackend.Panel/Views/Lessons/GroupSelection.cshtml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                            <tr>
+                                <th>
+                                    <span>Groups</span>
+                                </th>
+
+                                <th></th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            @foreach (var item in Model)
```
Comment (BobTheSuperman) [inline CharlieBackend.Panel/Views/Lessons/GroupSelection.cshtml]: Again "item"

**Review by BobTheSuperman — APPROVED** [review]
In my opinion, it is necessary to change the names of some variables.

File: CharlieBackend.Data/Configurations/VisitEntityConfiguration.cs [hunk 5]
```diff
@@ -27,7 +27,7 @@ public void Configure(EntityTypeBuilder<Visit> entity)
 
             entity.Property(e => e.Presence)
                 .IsRequired()
-                .HasDefaultValueSql("1")
+                //.HasDefaultValueSql("1")
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Data/Configurations/VisitEntityConfiguration.cs:30]: please explain why this is commented and if it is not needed then delete it.

File: CharlieBackend.Panel/Services/LessonService.cs [hunk 6]
```diff
@@ -1,29 +1,225 @@
...
+                Mentors = await mentors,
+                Themes = await themes,
+                Students = students,
+                LessonVisits = students.Select(s => new VisitDto { StudentId = s.Id }).ToList()
+            };
+
+            return newLesson;
+        }
+
+        public async Task<LessonDto> GetLessonById(long id)
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Panel/Services/LessonService.cs]: please add asyn everywhere if method is async.

File: CharlieBackend.Panel/Controllers/LessonsController.cs [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        public async Task<IActionResult> CreateLesson(long stGroupId, long? eventId)
+        {
+            if (User.IsInRole("Mentor"))
+            {
+                ViewBag.MentorId = _currentUserService.EntityId;
+            }
+            var LessonData = await _lessonService.PrepareLessonAddAsync(stGroupId);
+
+            ViewBag.EventId = eventId;
+
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Panel/Controllers/LessonsController.cs]: please remove space

File: CharlieBackend.Panel/Models/Mapping/ViewModelMapping.cs [hunk 8]
```diff
@@ -36,7 +38,31 @@ public ViewModelMapping()
...
+            CreateMap<LessonDto, LessonVisitModel>()
+               .ForMember(destination => destination.Visit, config => config.MapFrom(x => x.LessonVisits));
+
+            CreateMap<LessonDto, LessonUpdateViewModel>();
+
+            CreateMap<LessonUpdateViewModel, UpdateLessonDto>();
+
+            CreateMap<LessonCreateViewModel, CreateLessonDto>()
+                .ForMember(destination => destination.LessonVisits, config => config.MapFrom(x => x.LessonVisits.
+                Select(y => new  VisitDto
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Panel/Models/Mapping/ViewModelMapping.cs]: please remove select and fix mapping

File: CharlieBackend.Panel/Models/Mapping/ViewModelMapping.cs [hunk 9]
```diff
@@ -48,6 +74,9 @@ public ViewModelMapping()
 
             CreateMap<CourseDto, CourseViewModel>();
 
+            CreateMap<ScheduledEventViewModel, ScheduledEventDTO>();
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Panel/Models/Mapping/ViewModelMapping.cs:71]: please use reverse map

File: CharlieBackend.Panel/Services/EventsService.cs [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.Extensions.Options;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Panel.Services
+{
+    public class EventsService : IEventsService
+    {
+        private readonly EventsApiEndpoints _eventsApiEndpoints;
+        private readonly IApiUtil _apiUtil;
+        public EventsService(IOptions<ApplicationSettings> options,
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Panel/Services/EventsService.cs:14]: please add spaces where needed.

**Review by ArsenMaistryshyn — CHANGES_REQUESTED** [review]

**Review by ArsenMaistryshyn — APPROVED** [review]

**Pull request merged**